lib/animina_web/potential_partner.ex

```diff
@@ -213,4 +356,16 @@ defmodule AniminaWeb.PotentialPartner do
         []
     end
   end
+
+  def convert_to_date(date_string) do
+    {:ok, date} = Date.from_iso8601(date_string)
+    date
+  end
+
+  def calculate_age(birthdate) do
```

Comment:
This can lead to some edge cases. Please use something like this:

```elixir
def calculate_age(birthdate) do
  today = Date.utc_today()
  years = today.year - birthdate.year

  if Date.compare(today, ~D[#{birthdate.year + years}-#{birthdate.month}-#{birthdate.day}]) == :lt do
    years - 1
  else
    years
  end
end
```

And make sure that `birthdate` is not in the future. Example:

```elixir
def calculate_age(birthdate) when birthdate > Date.utc_today(), do: {:error, "Birthdate is in the future"}
```

Reply:
The next PR was to handle validations that are already there in the user resource for this page.